bmh_admin_portal_ui/src/components/admin-route.jsx

```diff
@@ -12,20 +12,39 @@ import {
 import { isAuthenticated } from "../util/oidc"
 
 const AdminRoute = ({ component: Component, ...rest }) => {
-  const [adminAuthorized, setAdminAuthorized] = useState(false);
+
+  const [adminAuthorized, setAdminAuthorized] = useState(null);
+
   useEffect(() => {
     async function fetchAuthorized() {
       const adminAuthorized = await authorizeAdmin();
       setAdminAuthorized(adminAuthorized);
     }
     fetchAuthorized();
   }, []);
+
   const isLoggedIn = isAuthenticated()
+  if (!isLoggedIn){
+    return (
+      <Route
+        {...rest}
+        render={props =>
+          <Redirect to={{ pathname: '/', state: { from: props.location } }} />
+        }
+      />
+    )
+  }
+
+  if(adminAuthorized===null){ //Only return <div/> if adminAuthorized equal to null, not when equal to false.
```

Comment:
```suggestion
  if(isLoggedIn && adminAuthorized===null){ //Only return <div/> if adminAuthorized equal to null, not when equal to false.
```
if you do this then you can merge the logic from L27-L36 into the block below?

Reply:
Yeah! That seemed like a nice idea. I just implemented and tested this change.